:100644 100644 64f27ef5af 0000000000 M	ext/standard/basic_functions.c
:100644 100644 339c45a6a8 0000000000 M	ext/standard/dns.c
:100644 100644 295751f0db 0000000000 M	ext/standard/php_fopen_wrapper.c
:100644 100644 29967df34a 0000000000 M	ext/standard/syslog.c

diff --git a/ext/standard/basic_functions.c b/ext/standard/basic_functions.c
index 64f27ef5af..e4d4cd2377 100644
--- a/ext/standard/basic_functions.c
+++ b/ext/standard/basic_functions.c
@@ -3736,7 +3736,7 @@ PHP_MINIT_FUNCTION(basic) /* {{{ */
 PHP_MSHUTDOWN_FUNCTION(basic) /* {{{ */
 {
 #ifdef HAVE_SYSLOG_H
-	PHP_MSHUTDOWN(syslog)(SHUTDOWN_FUNC_ARGS_PASSTHRU);
+	BASIC_MSHUTDOWN_SUBMODULE(syslog);
 #endif
 #ifdef ZTS
 	ts_free_id(basic_globals_id);
diff --git a/ext/standard/dns.c b/ext/standard/dns.c
index 339c45a6a8..ea8e4a1a44 100644
--- a/ext/standard/dns.c
+++ b/ext/standard/dns.c
@@ -116,6 +116,8 @@
 #endif
 /* }}} */
 
+#define dn_skipname(...) -1
+
 static zend_string *php_gethostbyaddr(char *ip);
 static zend_string *php_gethostbyname(char *name);
 
diff --git a/ext/standard/php_fopen_wrapper.c b/ext/standard/php_fopen_wrapper.c
index 295751f0db..1e74e38635 100644
--- a/ext/standard/php_fopen_wrapper.c
+++ b/ext/standard/php_fopen_wrapper.c
@@ -31,6 +31,8 @@
 #include "php_fopen_wrappers.h"
 #include "SAPI.h"
 
+#define getdtablesize(...) INT_MAX
+
 static ssize_t php_stream_output_write(php_stream *stream, const char *buf, size_t count) /* {{{ */
 {
 	PHPWRITE(buf, count);
diff --git a/ext/standard/syslog.c b/ext/standard/syslog.c
index 29967df34a..b178a849cc 100644
--- a/ext/standard/syslog.c
+++ b/ext/standard/syslog.c
@@ -106,13 +106,11 @@ PHP_RINIT_FUNCTION(syslog)
 }
 
 
-#ifdef PHP_WIN32
 PHP_RSHUTDOWN_FUNCTION(syslog)
 {
 	closelog();
 	return SUCCESS;
 }
-#endif
 
 PHP_MSHUTDOWN_FUNCTION(syslog)
 {
